core/config/config-helpers.js

```diff
@@ -120,7 +120,7 @@ const mergeConfigFragment = _mergeConfigFragment;
  * Merge an array of items by a caller-defined key. `mergeConfigFragment` is used to merge any items
  * with a matching key.
  *
- * @template T
+ * @template {Record<string, any>} T
  * @param {Array<T>|null|undefined} baseArray
```

Comment:
btw, the mergeConfigFragment has a comment about doing the typing in config.d.ts because jsodc does not support constraints. it does now (hence this change), but I didn't go further than fixing errors that cropped up.